src/swap-manager.ts

```
@@ -919,11 +919,20 @@ export class SwapManager {
                         `Auto-signing server's cooperative claim for ARK chain swap ${swap.id}`
                     );
-                    await this.executeSignServerClaimAction(swap);
-                    // Emit action executed event to all listeners
-                    this.actionExecutedListeners.forEach((listener) =>
-                        listener(swap, "signServerClaim")
-                    );
+                    try {
+                        const signed =
+                            await this.executeSignServerClaimAction(swap);
+                        if (signed) {
+                            this.actionExecutedListeners.forEach((listener) =>
```

Biome (2.4.4)

[error] This callback passed to forEach() iterable method should not return a value.

(lint/suspicious/useIterableCallbackReturn)